### PR TITLE
Update publish_v3.yml to alias "v3.0" (and later "v3.0.1") as "latest"

### LINKED ISSUE #949: https://spdx.github.io/spdx-spec/latest/ redirects to v2.3
The URL https://spdx.github.io/spdx-spec/latest/ is redirected to https://spdx.github.io/spdx-spec/v2.3/

The general expected behaviour for `latest/` is to redirected to the latest (stable) version available, which currently is `v3.0`.

### PR DESCRIPTION
**Pull request by bact**
`.github/workflows/publish_v3.yml`:

- Update `alias` argument for `mike deploy` to have `v3.0` as the `latest`
  - `v3.0` was aliased as `v3-draft`, so `v2.3` still occupied the `latest` alias.

`mkdocs.yml`:

- `mike` section: Add `canonical_version` configuration as `latest`
  - This will help the `latest` appear near the first in search engine, among other versions, see https://github.com/jimporter/mike#configuration 

Will fix #949

### CODE REVIEW

**Review by goneall — APPROVED** [review]
LGTM

**Comment by goneall** [comment]
@licquia - could you do a quick review

**Comment by goneall** [comment]
@bact @licquia - note the related PR #934 - can we combine these into one PR?

**Comment by bact** [comment]
@goneall @licquia Yes. If we don't plan to build docs for 3.0 again, I think it make sense to combine this PR with the PR for 3.0.1.

**Comment by bact** [comment]
@goneall Updated. Bring in changes for 3.0.1 by @licquia from PR #934 to here.

**Comment by goneall** [comment]
Thanks @bact - @licquia and/or @zvr - if you could do a quick review, we can merge

**Comment by bact** [comment]
I'm try to rethink and find what more logical is:

1. on `development/v3.0` branch, set `checkout` to use `3.0` tag
    - `publish_v3.yml`:
      ```yaml
      - uses: actions/checkout@f4...44 # v3
        with:
          repository: spdx/spdx-3-model
          ref: 3.0
          path: spdx-3-model
      ```
2. on `development/v3.0` branch, set `latest` alias to point to `v3.0` version
    - `publish_v3.yml`:
      ```yaml
      run: mike deploy v3.0 latest -b gh-pages -p
      ```
3. on `development/v3.0` branch, set `canonical_version` to `latest`
    - `mkdocs.yml`: 
      ```yaml
      - mike:
        canonical_version: latest
      ```
4. on `development/v3.0` branch, run the publish workflow to have the proper latest on the current stable release (`v3.0`) -- DONE for v3.0
5. on `development/v3.0.1` branch, set the version (for mkdocs) to `v3.0.1-draft` (or `v3.0.1-RCxxx`) (an alias can be added, if needed)
    - `publish_v3.yml`:
      ```yaml
      run: mike deploy v3.0.1-draft -b gh-pages -p
      ```
   - The CI will checkout from `main` branch (development branch)
   - The CI will publish the unfinished v3.01 to `v3.0.1-draft` version (selectable from dropdown list in HTML)
6. (once v3.0.1 is ready) on `spdx-3-model` repo, release it using the tag `3.0.1`
7. (once v3.0.1 is ready) on `development/v3.0.1` branch, set `checkout` to use `3.0.1` tag
    - `publish_v3.yml`:
      ```yaml
      - uses: actions/checkout@f4...44 # v3
        with:
          repository: spdx/spdx-3-model
          ref: 3.0.1
          path: spdx-3-model
      ```
8. (once v3.0.1 is ready) on `development/v3.0.1` branch, publish it with `3.0.1` version, with the alias `latest`
    - `publish_v3.yml`:
      ```yaml
      run: mike deploy v3.0.1 latest -b gh-pages -p
      ```
9. (once v3.0.1 is ready) on `development/v3.0.1` branch,  set `canonical_version` to `latest`
    - `mkdocs.yml`: 
      ```yaml
      - mike:
        canonical_version: latest
      ```
10. (once v3.0.1 is ready) on `development/v3.0.1` branch, run the publish workflow to have the proper latest on the current stable release (now `v3.0.1`)


The content of (2)-(3) is PR #950 (this PR) before the merge with #934 (this PR at the commit `d44e0a6`) but it should be on `development/v3.0` branch instead.



--

If this is correct and agreed:

- I should close this PR to and open a new pull request with content in (1)-(3) in `development/v3.0` branch. That new PR will be reviewed and merged before (4).
- PR #934 can be modified to be (5) and merge now, with the target of using it during the development period of v3.0.1
- Once v3.0.1 is ready, another PR with the content from (6)-(9) should be open and merge before (10)

**Review by licquia — APPROVED** [review]
With this change, we will be advertising 3.0.1 as the latest; if we want to change that, we can do that with a slight change and some manual runs of mike.  But I think we should go forward with this at least as a first step.

**Comment by bact** [comment]
- Note: We also need to decide on how to update the copyright information at the footer of the page as well. #930
  - It is in the `copyright:` section of `mkdocs.yml`

https://github.com/spdx/spdx-spec/blob/aac3e38b0b01b2ccb129367617a07598cf0ccdd8/mkdocs.yml#L2

**Comment by bact** [comment]
> With this change, we will be advertising 3.0.1 as the latest; if we want to change that, we can do that with a slight change and some manual runs of mike. But I think we should go forward with this at least as a first step.

@licquia  Also, this workflow need to be slightly adjusted once #978 and #979 are implemented - but I agree that we can take this PR as a first step and then having new PRs for new features.

File: .github/workflows/publish_v3.yml [hunk 1]
```diff
@@ -39,15 +39,15 @@ jobs:
           shacl2code generate \
             --input spdx-spec/docs/model/spdx-model.ttl \
             --input spdx-spec/docs/model/jsonld-annotations.ttl \
-            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0.0/spdx-context.jsonld \
+            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0.1/spdx-context.jsonld \
```
Comment (zvr) [inline .github/workflows/publish_v3.yml:42]:
```suggestion
            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0/spdx-context.jsonld \
```

**Review by zvr — CHANGES_REQUESTED** [review]
No third-level number in URL

**Review by zvr — APPROVED** [review]

File: .github/workflows/publish_v3.yml [hunk 2]
```diff
@@ -39,15 +39,15 @@ jobs:
           shacl2code generate \
             --input spdx-spec/docs/model/spdx-model.ttl \
             --input spdx-spec/docs/model/jsonld-annotations.ttl \
-            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0.0/spdx-context.jsonld \
+            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0/spdx-context.jsonld \
```
Comment (zvr) [inline .github/workflows/publish_v3.yml:42]:
```suggestion
            --context-url spdx-spec/docs/model/spdx-context.jsonld https://spdx.org/rdf/3.0.1/spdx-context.jsonld \
```

**Review by zvr — CHANGES_REQUESTED** [review]
always use 3 levels

**Comment by bact** [comment]
Discussed in 2024-07-16 Tech team meeting - ready to merge.

**Review by zvr — APPROVED** [review]

**Comment by bact** [comment]
The deployment was failed because v2.3 still occupied "latest" alias. Need to use `-u` option in `mike deploy` to override that.

- To assign alias `latest` to `v3.0`, use this PR #994 
- To publish v3.0.1 (development) as `v3.0.1-draft`, use this PR #995

**Pull request merged**